### PR TITLE
fast-dds: modernize for conan v2

### PR DESCRIPTION
**Pull request by tankeco**
Specify library name and version:  **fast-dds/2.3.***

modernize for conan v2

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by tankeco** [comment]
how to fix the Missing prebuilt package error

**Comment by prince-chrismc** [comment]
> foonathan-memory

I made https://github.com/conan-io/conan-center-index/pull/17024 :)

File: recipes/fast-dds/all/conanfile.py [hunk 1]
```diff
@@ -27,23 +41,12 @@ class FastDDSConan(ConanFile):
...
-    @property
-    def _minimum_cpp_standard(self):
+    def _min_cppstd(self):
         return 11
 
     @property
-    def _minimum_compilers_version(self):
+    def _compilers_minimum_version(self):
         return {
             "Visual Studio": "16",
```
Comment (prince-chrismc) [inline recipes/fast-dds/all/conanfile.py:51]: handled by check_min_vs

```suggestion
```

File: recipes/fast-dds/all/conanfile.py [hunk 2]
```diff
@@ -52,88 +55,87 @@ def _minimum_compilers_version(self):
...
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
                 )
-            )
-        if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
-            # This combination leads to an fast-dds error when linking
-            # linking dynamic '*.dll' and static MT runtime
-            raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```
Comment (prince-chrismc) [inline recipes/fast-dds/all/conanfile.py:91]: Why does this not work? seems odd it used to 🤔

File: recipes/fast-dds/all/conanfile.py [hunk 3]
```diff
@@ -153,20 +155,21 @@ def package_info(self):
 
         # component fastrtps
         self.cpp_info.components["fastrtps"].set_property("cmake_target_name", "fastrtps")
-        self.cpp_info.components["fastrtps"].libs = tools.collect_libs(self)
+        self.cpp_info.components["fastrtps"].set_property("cmake_target_aliases", ["fastdds::fastrtps"])
```
Comment (prince-chrismc) [inline recipes/fast-dds/all/conanfile.py:158]: what is the reason for adding an alias?

**Review by prince-chrismc — COMMENTED** [review]
This looks super good just a few small questions :)

File: recipes/fast-dds/all/conanfile.py (hunk 2 above)
Comment (tankeco) [reply recipes/fast-dds/all/conanfile.py]: fixed. I forgot to make changes after copying from the template.

File: recipes/fast-dds/all/conanfile.py (hunk 1 above)
Comment (tankeco) [reply recipes/fast-dds/all/conanfile.py]: fixed

File: recipes/fast-dds/all/conanfile.py (hunk 3 above)
Comment (tankeco) [reply recipes/fast-dds/all/conanfile.py:159]: It seems that it doesn't occur anywhere except in conan-center-index where 'fastdds::fastrtps' is found. However, it was aliased in the previous recipes, so I kept it.

**Review by RubenRBS — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**